rest/route/project.go

```diff
@@ -1048,3 +1048,66 @@ func (h *projectParametersGetHandler) Run(ctx context.Context) gimlet.Responder
 
 	return gimlet.NewJSONResponse(res)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PUT /rest/v2/projects/variables/rotate
+
+type projectVarsPutInput struct {
+	ToReplace   string `json:"to_replace"`
+	Replacement string `json:"replacement"`
+	DryRun      bool   `json:"dry_run"`
+}
+
+type projectVarsPutHandler struct {
+	replaceVars *projectVarsPutInput
+
+	sc       data.Connector
+	settings *evergreen.Settings
+}
+
+func makeProjectVarsPut(sc data.Connector, settings *evergreen.Settings) gimlet.RouteHandler {
+	return &projectVarsPutHandler{
+		sc:       sc,
+		settings: settings,
+	}
+}
+
+func (h *projectVarsPutHandler) Factory() gimlet.RouteHandler {
+	return &projectVarsPutHandler{
+		sc:       h.sc,
+		settings: h.settings,
+	}
+}
+
+// Parse fetches the project's identifier from the http request.
+func (h *projectVarsPutHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	replacements := &projectVarsPutInput{}
+	if err = json.Unmarshal(b, replacements); err != nil {
+		return errors.Wrap(err, "API error while unmarshalling JSON")
+	}
+	if replacements.ToReplace == "" {
+		return errors.New("'to_replace' parameter must be specified")
+	}
+	if replacements.Replacement == "" {
+		return errors.New("'replacement' parameter must be specified")
+	}
+	h.replaceVars = replacements
+	defer body.Close()
+	return nil
+}
+
+// Run updates a project by name.
```

Comment:
This comment is kind of confusing; I'd either put no comment or give more context.

Reply:
done